src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -82,6 +83,30 @@ protected override async Task CleanupConnections()
             }
         }
 
+        protected override Task DisconnectClientConnections(string instanceId)
+        {
+            if (string.IsNullOrEmpty(instanceId) || _connectionIds.Count == 0)
+            {
+                return Task.CompletedTask;
+            }
+            try
+            {
+                // Connected service instance is down, close client gracefully
+                var connections = _connectionIds.Where(c => c.Value == instanceId);
+                foreach (var connection in connections)
+                {
+                    var serviceMessage = new CloseConnectionMessage(connection.Key, errorMessage: "Service transient error.");
+                    _ = PerformDisconnectAsyncCore(connection.Key, false);
+                    _ = WriteAsync(serviceMessage);
```

Comment:
>WriteAsync [](start = 24, length = 10)

this is write back to the service so no need?

Reply:
I think it writes to clients, isn't it?

---
In reply to: [306695578](https://github.com/Azure/azure-signalr/pull/612#discussion_r306695578) [](ancestors = 306695578)